File: Hard/2366. Minimum Replacements to Sort the Array/2366. Minimum Replacements to Sort the Array.java
class Solution {
    public long minimumReplacement(int[] nums) {
        long operations = 0;
        int prevBound = nums[nums.length - 1];
        
        for (int i = nums.length - 2; i >= 0; i--) {
            int num = nums[i];
            int noOfTimes = (num + prevBound - 1) / prevBound;
            operations += noOfTimes - 1;
            prevBound = num / noOfTimes;
        }
        
        return operations;
    }
}

File: Hard/2366. Minimum Replacements to Sort the Array/2366. Minimum Replacements to Sort the Array.rs
impl Solution {
    pub fn minimum_replacement(nums: Vec<i32>) -> i64 {
        let mut operations = 0;
        let mut prev_bound = nums[nums.len() - 1];

        for &num in nums.iter().rev().skip(1) {
            let no_of_times = (num + prev_bound - 1) / prev_bound;
            operations += (no_of_times - 1) as i64;
            prev_bound = num / no_of_times;
        }

        operations
    }
}
